### PR TITLE
Issue #2203 Provide module for using GlobalWebAppConfig with Deployer

### PR DESCRIPTION
**Pull request by janbartel**
Make it easier for users to apply standard config to all webapps that they deployer by using the GlobalWebAppConfig binding with the deployer. #2203

### CODE REVIEW

File: jetty-deploy/src/main/config/etc/global-webapp-deploy.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+<Configure id="Server" class="org.eclipse.jetty.server.Server">
+  <Ref refid="DeploymentManager">
+    <Call name="addLifeCycleBinding">
+      <Arg>
+        <New class="org.eclipse.jetty.deploy.bindings.GlobalWebappConfigBinding">
+          <Set name="jettyXml">
+            <Property>
+              <Name>jetty.deploy.globalWebappConfig</Name>
+              <Default>
+                <Property name="jetty.base"/>/etc/global-webapp-config.xml
```
Comment (gregw) [inline jetty-deploy/src/main/config/etc/global-webapp-deploy.xml]: A template `global-webapp-config.xml` should be created in `jetty-deploy/src/main/config/modules/globalwebapp.d/global-webapp-config.xml` and copied into jetty-base/etc by the module activation.  In fact `global-webapp-deploy.xml` should also be copied into place in the same way (ie don't pollute etc with config files that are not used).

Comment (janbartel) [reply jetty-deploy/src/main/config/etc/global-webapp-deploy.xml]: I changed the name of the module to deployer-webapp-common, and the xml file for the Deployer config to be deployer-webapp-common.xml and added an empty config file for the webapp called webapp-common.xml. Both are copied into $jetty.base/etc.

Not sure I agree with the practice of copying $jetty.home xml files that should never be modified into $jetty.base/etc, as this just encourages users to modify them and then it makes upgrades more complicated.

**Comment by WalkerWatch** [comment]
I am not a huge fan of the names `deployer-webapp-common.mod` and `deployer-webapp-common.xml`. It just feels clunky to me. What if we did `global-webapp-common.xml` and `global-webapp-common.mod` instead? I might just be splitting hairs though.

**Comment by janbartel** [comment]
@WalkerWatch I originally had "global" in the names, but replaced it with "deployer" to clarify the linkage with the configuration of the deployer.  But, I'm happy to go with your recommendations.

**Pull request merged**